tests/unit/test_charm.py

```diff
@@ -316,163 +307,104 @@ def test_unit_ips(self):
-        leader unit must not reconfigure.
+        Tests the scenario that if current replica set hosts are not ready, the leader goes into
+        WaitingStatus and no attempt to reconfigure is made.
         """
         # preset values
         self.harness.set_leader(True)
         connection.return_value.__enter__.return_value.is_ready = False
+        connection.return_value.__enter__.return_value.get_replset_members.return_value = {
+            "1.1.1.1"
+        }
 
         # simulate 2nd MongoDB unit
         rel = self.harness.charm.model.get_relation("mongodb")
         self.harness.add_relation_unit(rel.id, "mongodb/1")
         self.harness.update_relation_data(rel.id, "mongodb/1", PEER_ADDR)
 
-        # verify that we do not reconfigure replica set
-        mongo_reconfigure.assert_not_called()
-
-    @patch_network_get(private_address="1.1.1.1")
-    @patch("charm.MongoDBConnection")
-    @patch("charm.MongodbOperatorCharm._need_replica_set_reconfiguration")
-    @patch("mongod_helpers.MongoDB.reconfigure_replica_set")
-    @patch("mongod_helpers.MongoDB.all_replicas_ready", return_value=True)
-    def test_mongodb_relation_joined_peer_ready(
-        self, all_replicas_ready, mongodb_reconfigure, need_reconfiguration, connection
-    ):
-        """Test leader unit operations when peer unit is ready to join the replica set.
-
-        Verifies that when a new peer is ready to join the replica set hosts get updated
-        accordingly and reconfigured if necessary.
-        """
-        # preset values
-        self.harness.set_leader(True)
-        connection.return_value.__enter__.return_value.is_ready = True
-
-        # test cases where replica set should and shouldn't be reconfigured
-        for reconfiguration in [False, True]:
-            self.harness.charm._need_replica_set_reconfiguration = reconfiguration
-
-            # simulate 2nd MongoDB unit
-            rel = self.harness.charm.model.get_relation("mongodb")
-            self.harness.add_relation_unit(rel.id, "mongodb/1")
-            self.harness.update_relation_data(rel.id, "mongodb/1", PEER_ADDR)
-
-            # check if mongod reconfigured replica set
-            if reconfiguration:
-                mongodb_reconfigure.assert_called()
-
-                # verify replica set hosts updated accordingly
-                self.assertEqual(
-                    self.harness.charm._replica_set_hosts,
-                    ["127.4.5.6", "1.1.1.1"],
-                )
-
-            else:
-                mongodb_reconfigure.assert_not_called()
+        # verify we go into waiting and don't reconfigure
+        self.assertTrue(isinstance(self.harness.charm.unit.status, WaitingStatus))
+        connection.return_value.__enter__.return_value.add_replset_member.assert_not_called()
 
     @patch_network_get(private_address="1.1.1.1")
+    @patch("ops.framework.EventBase.defer")
```

Comment:
Not completely related to the PR.

With imports from the charm module you could potentially keep it here:

```python
from ops.framework import EventBase
. . .
# tests
. . .
@patch.object(EventBase, "defer")
def test_relation_joined_get_members_failure(defer):
```

Or in places that you are using `@patch("ops.model.ActiveStatus")`. Since `ActiveStatus` is being imported at the top already, you can just do:
```python
@patch.object(ActiveStatus)
def test(active_status):
```